products/editions/enterprise-edition/b2b-suite/guides/core/assignment-service.md

```diff
@@ -0,0 +1,89 @@
+# Assignment service
+
+## Table of contents
+*   [The Pattern](#the-pattern)
+*   [The Repository](#the-repository)
+*   [The Service](#the-service)
+
+## The Pattern
+A repeating pattern used throughout the B2B-Suite are Assignment Services. The B2B-Suite ships with many new entities, and therefore provides the means to connect them to each other. This is done through M:N assignments in for that purpose alone created components.
+
+The Diagram below shows the usually implemented objects with their outside dependencies.
+
+![image](/.gitbook/assets/assignment-service.svg)
+
+## The Repository
+Again the repository is the exclusive access layer to the storage engine. Contrary to CRUD operations there is no object, but just plain integers (The primary keys). The default repository will have these three methods relevant for assignment:
+
+```php
+<?php
+
+namespace Shopware\B2B\RoleContact\Framework;
+
+use Doctrine\DBAL\Connection;
+use Shopware\B2B\Common\Repository\DbalHelper;
+
+class RoleContactRepository
+{
+    /**
+     * @param int $roleId
```

Comment:
I think the param and return annotations are redundant. It would be much cleaner if you remove them :)